### PR TITLE
Support for sparse images from Halium-7.1 builds

### PR DESCRIPTION
**Pull request by Flohack74**
This will allow support for sparse images created by the Halium-7.1 builds, first one here is suzu by @fredldotme :)

### CODE REVIEW

File: build-tarball.sh [hunk 1]
```diff
@@ -29,6 +29,18 @@ cp $dir/recovery.img $wDir/partitions
...
 mkdir -p $wDir/system/var/lib/lxc/android/
+
+## SPARSE FILE TRANSLATION
+# Needed with Halium-7.1 builds mostly
+isSparse=$(file -b0 $dir/system.img)
+if [[ $isSparse == "Android sparse image"* ]]; then
+    echo "Converting sparse image to image"
+    mv $dir/system.img $dir/system.sparse.img
+    simg2img $dir/system.sparse.img $dir/system.img
+    e2fsck -fy $dir/system.img >/dev/null
```
Comment (UniversalSuperBox) [inline build-tarball.sh:40]: Do you mean `&>/dev/null`? This will only send stdout, not stderr, to null.

File: build-tarball.sh [hunk 2]
```diff
@@ -29,6 +29,18 @@ cp $dir/recovery.img $wDir/partitions
 cp -r "$device_dir/ubuntu-overlay/*" $wDir/ || true
 
 mkdir -p $wDir/system/var/lib/lxc/android/
+
+## SPARSE FILE TRANSLATION
+# Needed with Halium-7.1 builds mostly
+isSparse=$(file -b0 $dir/system.img)
+if [[ $isSparse == "Android sparse image"* ]]; then
```
Comment (UniversalSuperBox) [inline build-tarball.sh]: I question the name of the variable isSparse. Maybe `systemFileType` would make more sense. `isSparse` implies its value is boolean.

File: build-tarball.sh (hunk 1 above)
Comment (Flohack74) [reply build-tarball.sh:40]: Huh what do I mean I have to think about it ^^

Comment (Flohack74) [reply build-tarball.sh:40]: I think better leave errors in the log. That could show potential problems better.

**Review by UniversalSuperBox — APPROVED** [review]

**Pull request merged**